src/lib.rs

```diff
@@ -356,7 +356,7 @@ lazy_static! {
 }
 
 /// Internal representation of a weighted task list.
-type WeightedGooseTasks = Vec<Vec<usize>>;
+type WeightedGooseTasks = Vec<Vec<(usize, String)>>;
```

Comment:
This is the main change in this PR, expanding `WeightedGooseTasks` to also include a the GooseTask name, if set.